test/unit/ngsiv2/provisioning/singleConfigurationMode-test.js

```diff
@@ -0,0 +1,333 @@
+        }
+    },
+    deviceCreation = {
+        url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+        method: 'POST',
+        json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionNewDevice.json'),
+        headers: {
+            'fiware-service': 'TestService',
+            'fiware-servicepath': '/testingPath'
+        }
+    };
+
+describe('Provisioning API: Single service mode', function() {
+    beforeEach(function(done) {
+        nock.cleanAll();
+
+        iotAgentLib.activate(iotAgentConfig, function() {
+            iotAgentLib.clearAll(done);
+        });
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+        iotAgentLib.setProvisioningHandler();
+        iotAgentLib.deactivate(done);
+    });
+
+    describe('When a new configuration arrives to an already configured subservice', function() {
+        var groupCreationDuplicated = {
+            url: 'http://localhost:4041/iot/services',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/examples/groupProvisioningRequests/provisionDuplicateGroup.json'),
+            headers: {
+                'fiware-service': 'TestService',
+                'fiware-servicepath': '/testingPath'
+            }
+        };
+
+        beforeEach(function(done) {
+            request(groupCreation, done);
+        });
+
+        it('should raise a DUPLICATE_GROUP error', function(done) {
+            request(groupCreationDuplicated, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(409);
+                should.exist(body.name);
+                body.name.should.equal('DUPLICATE_GROUP');
+                done();
+            });
+        });
+    });
+    describe('When a device is provisioned with an ID that already exists in the configuration', function() {
+        var deviceCreationDuplicated = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionDuplicatedDev.json'),
+            headers: {
+                'fiware-service': 'TestService',
+                'fiware-servicepath': '/testingPath'
+            }
+        };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5